app/src/main/java/com/randomappsinc/studentpicker/home/MainActivity.java

```diff
@@ -310,4 +323,42 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+
+    private void showRenameDialog(final int position) {
+        new MaterialDialog.Builder(this)
+                .title(R.string.rename_list)
+                .input(getString(R.string.new_list_name), "", (dialog, input) -> {
+                    boolean submitEnabled = !(input.toString().trim().isEmpty() ||
+                            preferencesManager.doesListExist(input.toString().trim()));
+                    dialog.getActionButton(DialogAction.POSITIVE).setEnabled(submitEnabled);
+                })
+                .alwaysCallInputCallback()
+                .negativeText(android.R.string.cancel)
+                .onAny((dialog, which) -> {
+                    if (which == DialogAction.POSITIVE) {
+                        String newListName = dialog.getInputEditText().getText().toString().trim();
+                        dataSource.renameList(nameListsAdapter.getItem(position), newListName);
+                        preferencesManager.renameList(nameListsAdapter.getItem(position), newListName);
+                        nameListsAdapter.getContent().set(position, newListName);
+                        nameListsAdapter.notifyItemChanged(position);
+                    }
+                })
+                .show();
+    }
+
+    private void showDeleteDialog(final int position) {
+        new MaterialDialog.Builder(this)
+                .title(R.string.confirm_deletion_title)
+                .content(R.string.confirm_deletion_message)
+                .positiveText(android.R.string.yes)
+                .negativeText(android.R.string.cancel)
+                .onPositive((dialog, which) -> {
+                    dataSource.deleteList(nameListsAdapter.getItem(position));
+                    preferencesManager.removeNameList(nameListsAdapter.getItem(position));
+                    nameListsAdapter.getContent().remove(position);
```

Comment:
nit: Same as above, let's expose a method in the adapter called "deleteItem(int position)".